File: python-service/app/schemas/order.py
from pydantic import BaseModel, Field

from app.schemas.common import InternalRequestMeta


class CaregiverSnapshotDTO(BaseModel):
    user_id: str
    nickname: str
    avatar: str | None = None
    cat_care_score: float | None = None
    tags: list[str] = Field(default_factory=list)
    supports_medication: bool | None = None
    supports_multi_day_care: bool | None = None


class DriverSnapshotDTO(BaseModel):
    user_id: str
    nickname: str
    avatar: str | None = None
    pet_friendly_score: float | None = None
    tags: list[str] = Field(default_factory=list)
    vehicle_type: str | None = None


class OrderSnapshotRequest(BaseModel):
    meta: InternalRequestMeta
    order_id: str
    provider_user_id: str
    vehicle_id: str | None = None


class OrderSnapshotResponse(BaseModel):
    request_id: str
    order_id: str
    caregiver_snapshot: CaregiverSnapshotDTO | None = None
    driver_snapshot: DriverSnapshotDTO | None = None

File: shared/order.ts
import type { LocationPoint } from './location.js';

export type OfferStatus = 'submitted' | 'accepted' | 'rejected' | 'withdrawn';
export type OrderStatus =
  | 'pending_payment'
  | 'paid'
  | 'confirmed'
  | 'arriving'
  | 'serving'
  | 'completed'
  | 'cancelled'
  | 'refund_pending'
  | 'refunded';

export interface Offer {
  id: string;
  demandId: string;
  providerUserId: string;
  quoteAmountFen: number;
  message?: string;
  etaMinutes?: number;
  vehicleId?: string;
  servicePlan?: string;
  status: OfferStatus;
  createdAt: string;
}

export interface DriverSnapshot {
  userId: string;
  nickname: string;
  avatar: string;
  petFriendlyScore?: number;
  tags: string[];
  vehicleType?: string;
}

export interface CaregiverSnapshot {
  userId: string;
  nickname: string;
  avatar: string;
  catCareScore?: number;
  tags: string[];
  supportsMedication?: boolean;
  supportsMultiDayCare?: boolean;
}

export interface ServiceOrder {
  id: string;
  demandId: string;
  buyerUserId: string;
  sellerUserId: string;
  title: string;
  amountFen: number;
  depositFen?: number;
  status: OrderStatus;
  serviceTime: string;
  pickup?: LocationPoint;
  destination?: LocationPoint;
  vehicleId?: string;
  driverSnapshot?: DriverSnapshot;
  caregiverSnapshot?: CaregiverSnapshot;
  createdAt: string;
  updatedAt: string;
}

export interface OrderEvent {
  id: string;
  orderId: string;
  eventType: string;
  operatorUserId?: string;
  payload?: Record<string, unknown>;
  createdAt: string;
}
